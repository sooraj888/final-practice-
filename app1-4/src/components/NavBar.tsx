import { AppBar, Toolbar, Typography } from "@material-ui/core";
import React from "react";

const NavBar = (): JSX.Element => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
