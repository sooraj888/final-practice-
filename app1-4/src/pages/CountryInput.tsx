import { Box, Button, TextField } from "@material-ui/core";
import React, { FormEvent, useState } from "react";
import { NavigateFunction, useNavigate } from "react-router-dom";

const CountryInput = (): JSX.Element => {
  const navigate: NavigateFunction = useNavigate();
  const [countryName, setCountryName] = useState<string>("");

  const handleOnSubmit = (e: FormEvent<HTMLFormElement>): void => {
    e.preventDefault();
    navigate(`/${countryName}`);
  };
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        width: "200px",
        height: "300px",
        justifyContent: "space-around",
        padding: "30px",
        border: "3px solid gray",
        margin: "30px",
        borderRadius: "20px",
      }}
      component={"form"}
      onSubmit={handleOnSubmit}
    >
      <TextField
        value={countryName}
        onChange={(e: any) => {
          setCountryName(e.target.value);
        }}
        required
        variant="outlined"
        placeholder="Enter Country name"
      ></TextField>
      <Button variant="contained" color="primary" type="submit">
        Search
      </Button>
    </Box>
  );
};

export default CountryInput;
