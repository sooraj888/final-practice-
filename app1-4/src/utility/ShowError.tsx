import { Box } from "@material-ui/core";
import React from "react";

const ShowError = (): JSX.Element => {
  return (
    <Box>
      <h1>
        <div>Oops</div>
        Somthing Went wrong!
      </h1>
    </Box>
  );
};

export default ShowError;
