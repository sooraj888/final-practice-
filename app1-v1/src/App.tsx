import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Box } from "@material-ui/core";

function App() {
  return <Box className="App"></Box>;
}

export default App;
