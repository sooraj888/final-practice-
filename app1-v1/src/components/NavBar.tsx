import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyles = makeStyles({
  roots: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.roots}>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
