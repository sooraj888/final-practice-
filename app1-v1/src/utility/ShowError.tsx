import React from "react";

const ShowError = (): JSX.Element => {
  return (
    <div>
      <h1>
        <p>oops</p>
        Somthing Went Weong
      </h1>
    </div>
  );
};

export default ShowError;
