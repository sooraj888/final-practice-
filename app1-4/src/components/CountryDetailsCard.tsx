import {
  Box,
  Button,
  Card,
  CardContent,
  CardMedia,
  Typography,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React, { useEffect, useState } from "react";
import { Navigate, NavigateFunction, useNavigate } from "react-router-dom";
import useApiFetch from "../hooks/useApiFetch";
import ShowError from "../utility/ShowError";

const useStyles = makeStyles({
  title: {
    fontWeight: 700,
    color: "gray",
  },
  subTitle: { fontWeight: 900 },
});

export type useApiFetchType = {
  data: any;
  isLoading: boolean;
  error: boolean;
};
const CountryDetailsCard = ({ countryData }: any): JSX.Element => {
  const classes = useStyles();
  const navigate: NavigateFunction = useNavigate();
  const url: string =
    "http://api.weatherstack.com/current?access_key=dbd5b5a511521f50e7b63d9be7d96047&query=";
  const { data, error }: useApiFetchType = useApiFetch(
    url,
    countryData?.capital
  );
  const [showWeather, setShowWeather] = useState<boolean>(false);

  useEffect((): void => {
    console.log(data);
  }, [data]);
  return (
    <Card>
      {!showWeather ? (
        <Box
          sx={{
            display: "flex",
            padding: "30px",
            flexDirection: "column",
            alignItems: "center",
          }}
        >
          <CardMedia component={"img"} image={countryData?.flags?.png} />
          <CardContent>
            <Typography className={classes.title}>Country</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.name?.common}
            </Typography>
            <Typography className={classes.title}>Capital</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.capital}
            </Typography>
            <Typography className={classes.title}>Population</Typography>
            <Typography className={classes.subTitle}>
              {countryData?.population}
            </Typography>
            <Typography className={classes.title}>Lat-Lng</Typography>
            <Typography>
              {countryData?.latlng?.[0]}-{countryData?.latlng?.[1]}
            </Typography>
          </CardContent>
          <Button
            variant="contained"
            color="primary"
            onClick={() => {
              setShowWeather(true);
            }}
          >
            Show Weather Data
          </Button>
        </Box>
      ) : (
        <>
          {error ? (
            <ShowError />
          ) : (
            <Box
              sx={{
                display: "flex",
                padding: "30px",
                flexDirection: "column",
                alignItems: "center",
              }}
            >
              <Box
                sx={{
                  display: "flex",
                  padding: "30px",
                }}
              >
                <CardContent>
                  <Typography className={classes.title}>Region</Typography>
                  <Typography className={classes.subTitle}>
                    {data?.location?.region}
                  </Typography>
                  <Typography className={classes.title}>Temperature</Typography>
                  <Typography className={classes.subTitle}>
                    {data?.current?.temperature}
                  </Typography>
                  <Typography className={classes.title}>Wind_speed:</Typography>
                  <Typography className={classes.subTitle}>
                    {data?.current?.wind_speed}
                  </Typography>
                </CardContent>

                <Box>
                  <CardMedia
                    component={"img"}
                    image={data?.current?.weather_icons?.[0]}
                    alt="flag"
                  />
                  <Typography className={classes.subTitle}>
                    {data?.current?.weather_descriptions}
                  </Typography>
                </Box>
              </Box>
              <Button
                variant="contained"
                color="primary"
                onClick={(): void => {
                  navigate("/");
                }}
              >
                Search Another country
              </Button>
            </Box>
          )}
        </>
      )}
    </Card>
  );
};

export default CountryDetailsCard;
