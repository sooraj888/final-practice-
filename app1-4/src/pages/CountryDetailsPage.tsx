import { Box, CircularProgress } from "@material-ui/core";
import React, { useEffect } from "react";
import { useLocation, useParams } from "react-router-dom";
import CountryDetailsCard, {
  useApiFetchType,
} from "../components/CountryDetailsCard";
import useApiFetch from "../hooks/useApiFetch";
import ShowError from "../utility/ShowError";

const CountryDetailsPage = (): JSX.Element => {
  const url: string = "https://restcountries.com/v3.1/name/";
  const countryName: string | undefined = useParams().name;
  const { data, error, isLoading }: useApiFetchType = useApiFetch(
    url,
    countryName
  );
  useEffect((): void => {
    console.log(data[0]);
  }, [data]);
  return (
    <Box sx={{ display: "flex", padding: "30px" }}>
      {error ? (
        <ShowError></ShowError>
      ) : isLoading ? (
        <>
          <CircularProgress />
          <p>Loading...</p>
        </>
      ) : data ? (
        <CountryDetailsCard countryData={data?.[0]}></CountryDetailsCard>
      ) : (
        <ShowError />
      )}
    </Box>
  );
};

export default CountryDetailsPage;
