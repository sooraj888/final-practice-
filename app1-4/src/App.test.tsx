import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import ShowError from "./utility/ShowError";
import NavBar from "./components/NavBar";
import CountryInput from "./pages/CountryInput";
import { BrowserRouter } from "react-router-dom";
import CountryDetailsPage from "./pages/CountryDetailsPage";
import CountryDetailsCard from "./components/CountryDetailsCard";

describe("test on ShowError componet ", () => {
  test("rendering ShowError componet on terminal", () => {
    render(<ShowError />);
    screen.debug();
  });
});

describe("test on ShowError componet ", () => {
  test("finding the element", () => {
    render(<ShowError />);
    const ele = screen.getByText(/Somthing Went wrong!/);
    expect(ele).toBeInTheDocument();
  });
});

describe("test on ShowError componet ", () => {
  test("snapshot tes on ShowError componet ", () => {
    const { container } = render(<ShowError />);
    expect(container).toMatchSnapshot();
  });
});

describe("test on NavBar componet ", () => {
  test("rendering NavBar componet on terminal", () => {
    render(<NavBar />);
    screen.debug();
  });
});

describe("test on NavBar componet ", () => {
  test("finding the element", () => {
    render(<NavBar />);
    const ele = screen.getByText(/Country Weather App/);
    expect(ele).toBeInTheDocument();
  });
});

describe("test on NavBar componet ", () => {
  test("snapshot tes on NavBar componet ", () => {
    const { container } = render(<NavBar />);

    expect(container).toMatchSnapshot();
  });
});

describe("test on CountryInput componet ", () => {
  test("rendering CountryInput componet on terminal", () => {
    render(
      <BrowserRouter>
        <CountryInput />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on CountryInput componet ", () => {
  test("finding the element", () => {
    render(
      <BrowserRouter>
        <CountryInput />
      </BrowserRouter>
    );
    const ele = screen.getByText(/Search/);
    expect(ele).toBeInTheDocument();
  });
});

describe("test on CountryInput componet ", () => {
  test("finding the element with role", () => {
    render(
      <BrowserRouter>
        <CountryInput />
      </BrowserRouter>
    );
    const ele = screen.getByRole("button");
    expect(ele).toBeInTheDocument();
  });
});

describe("test on CountryInput componet ", () => {
  test("snapshot tes on CountryInput componet ", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryInput />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});

describe("test on CountryInput componet ", () => {
  test(" snapshot tes and fireEvent changing input value ", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryInput />
      </BrowserRouter>
    );
    const ele = screen.getByRole("textbox");
    fireEvent.change(ele, { target: { value: "sooraj" } });
    expect(screen.getByDisplayValue("sooraj")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});

const countryData = {
  capital: "Dehli",
  latlng: [20, 17],
  population: "130cr",
  name: { common: "India" },
};

describe("test on CountryDetailsCard componet ", () => {
  test("rendering CountryDetailsCard componet on terminal without prop", () => {
    render(
      <BrowserRouter>
        <CountryDetailsCard />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on CountryDetailsCard componet ", () => {
  test("rendering CountryDetailsCard componet on terminal with prop", () => {
    render(
      <BrowserRouter>
        <CountryDetailsCard countryData={countryData} />
      </BrowserRouter>
    );
    screen.debug();
  });
});

describe("test on CountryDetailsCard componet ", () => {
  test("finding the element", () => {
    render(
      <BrowserRouter>
        <CountryDetailsCard countryData={countryData} />
      </BrowserRouter>
    );
    const ele = screen.getByText(/130cr/);
    expect(ele).toBeInTheDocument();
    expect(screen.getByText(/20-17/)).toBeInTheDocument();
    expect(screen.getByText(/Dehli/)).toBeInTheDocument();
    expect(screen.getByText(/India/)).toBeInTheDocument();
  });
});

describe("test on CountryDetailsCard componet ", () => {
  test("finding the element with role", () => {
    render(
      <BrowserRouter>
        <CountryDetailsCard />
      </BrowserRouter>
    );
    const ele = screen.getByRole("button");
    expect(ele).toBeInTheDocument();
  });
});

describe("test on CountryDetailsCard componet ", () => {
  test("snapshot tes on CountryInput componet  without prop", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryDetailsCard />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});

describe("test on CountryDetailsCard componet ", () => {
  test("snapshot test on CountryInput componet  with prop", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryDetailsCard countryData={countryData} />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
