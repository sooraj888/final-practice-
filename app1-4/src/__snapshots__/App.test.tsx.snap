// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on CountryDetailsCard componet  snapshot tes on CountryInput componet  without prop 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-43"
    >
      <img
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-title-41 MuiTypography-body1"
        >
          Country
        </p>
        <p
          class="MuiTypography-root makeStyles-subTitle-42 MuiTypography-body1"
        />
        <p
          class="MuiTypography-root makeStyles-title-41 MuiTypography-body1"
        >
          Capital
        </p>
        <p
          class="MuiTypography-root makeStyles-subTitle-42 MuiTypography-body1"
        />
        <p
          class="MuiTypography-root makeStyles-title-41 MuiTypography-body1"
        >
          Population
        </p>
        <p
          class="MuiTypography-root makeStyles-subTitle-42 MuiTypography-body1"
        />
        <p
          class="MuiTypography-root makeStyles-title-41 MuiTypography-body1"
        >
          Lat-Lng
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          -
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Show Weather Data
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`test on CountryDetailsCard componet  snapshot test on CountryInput componet  with prop 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-46"
    >
      <img
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-title-44 MuiTypography-body1"
        >
          Country
        </p>
        <p
          class="MuiTypography-root makeStyles-subTitle-45 MuiTypography-body1"
        >
          India
        </p>
        <p
          class="MuiTypography-root makeStyles-title-44 MuiTypography-body1"
        >
          Capital
        </p>
        <p
          class="MuiTypography-root makeStyles-subTitle-45 MuiTypography-body1"
        >
          Dehli
        </p>
        <p
          class="MuiTypography-root makeStyles-title-44 MuiTypography-body1"
        >
          Population
        </p>
        <p
          class="MuiTypography-root makeStyles-subTitle-45 MuiTypography-body1"
        >
          130cr
        </p>
        <p
          class="MuiTypography-root makeStyles-title-44 MuiTypography-body1"
        >
          Lat-Lng
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          20
          -
          17
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Show Weather Data
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`test on CountryInput componet   snapshot tes and fireEvent changing input value  1`] = `
<div>
  <form
    class="MuiBox-root MuiBox-root-24"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          placeholder="Enter Country name"
          required=""
          type="text"
          value="sooraj"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-25 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-26"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;

exports[`test on CountryInput componet  snapshot tes on CountryInput componet  1`] = `
<div>
  <form
    class="MuiBox-root MuiBox-root-19"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          placeholder="Enter Country name"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-20 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-21"
            style="width: 0.01px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;

exports[`test on NavBar componet  snapshot tes on NavBar componet  1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Country Weather App
      </p>
    </div>
  </header>
</div>
`;

exports[`test on ShowError componet  snapshot tes on ShowError componet  1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3"
  >
    <h1>
      <div>
        Oops
      </div>
      Somthing Went wrong!
    </h1>
  </div>
</div>
`;
